### PR TITLE
Run cargo test on Cirrus

### PR DESCRIPTION
**Pull request by k0kubun**
* Add `cargo test` on Cirrus
* Add some other Cirrus improvements
  * Drop `make rdoc` because that one doesn't seem to test YJIT
  * Replace `make btest` with `make test` to test more

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -126,8 +126,8 @@ yjit_task:
   test_dump_insns_script: RUST_BACKTRACE=1 ./miniruby --yjit-call-threshold=1 --yjit-dump-insns -e0
   output_stats_script: RUST_BACKTRACE=1 ./miniruby --yjit-call-threshold=1 --yjit-stats -e0
   full_build_script: source $HOME/.cargo/env && make -j
-  make_rdoc_script: source $HOME/.cargo/env && make -j rdoc
-  make_btest_script: source $HOME/.cargo/env && make -j btest RUN_OPTS="--yjit-call-threshold=1 --yjit-verify-ctx"
+  cargo_test_script: source $HOME/.cargo/env && cd yjit && cargo test
+  make_btest_script: source $HOME/.cargo/env && make -j test RUN_OPTS="--yjit-call-threshold=1 --yjit-verify-ctx"
```
Comment (maximecb) [inline .cirrus.yml:130]: Here we're already running `test-all` and `test-spec`

Is `make test` a superset of that?

**Comment by maximecb** [comment]
Let's add Kevin's change if it's needed to pass the extra tests.

File: .cirrus.yml (hunk 1 above)
Comment (k0kubun) [reply .cirrus.yml:130]: It's like:

* `make check`
  * `make test`
     * `make btest`
     * `make test-basic`
  * `make test-all`
  * `make test-spec`

I'm open to just using `make check` if we want to make YMAL shorter, but I feel like this level of separation makes it easier to see what's happening when it fails.

**Comment by k0kubun** [comment]
sure https://github.com/ruby/ruby/pull/6304/commits/67cbfce824a46c866d40ecc381df64bbef006d05

File: .cirrus.yml (hunk 1 above)
Comment (maximecb) [reply .cirrus.yml:130]: Yes I agree, I prefer having things separated out as well 👍

Comment (maximecb) [inline .cirrus.yml:130]:
```suggestion
  make_test_script: source $HOME/.cargo/env && make -j test RUN_OPTS="--yjit-call-threshold=1 --yjit-verify-ctx"
```

**Review by maximecb — APPROVED** [review]

**Review by XrXr — APPROVED** [review]

**Pull request merged**